libs/vkd3d/command.c

```diff
@@ -7165,15 +7167,15 @@ static void d3d12_command_list_barrier_batch_add_layout_transition(
     if (batch->image_barrier_count == ARRAY_SIZE(batch->vk_image_barriers))
         d3d12_command_list_barrier_batch_end(list, batch);
 
-    /* ResourceBarrier() in D3D12 behaves as if each transition happens in order.
-     * Vulkan memory barriers do not, so if there is a race condition, we need to split
-     * the barrier. */
-    for (i = 0; i < batch->image_barrier_count; i++)
-    {
-        if (vk_image_barrier_overlaps_subresource(image_barrier, &batch->vk_image_barriers[i]))
-        {
-            d3d12_command_list_barrier_batch_end(list, batch);
-            break;
+    if (batch->check_alias) {
```

Comment:
The check_alias is not very obvious to reason about. I can see this being a thing for something like src_resource in a copy. If we're trying to record a common layout -> TRANSFER_SRC_OPTIMAL all the time, we only really need to do that once. It might make more sense to just eliminate redundant barriers instead, e.g. same old/newLayout on same subresource range is just dropped.

For common layout -> TRANSFER_DST_OPTIMAL. If that is recorded multiple times, it means we have overlapped writes, which needs to be resolved as a WAW hazard either way. It might be better to handle that outside when emitting the copy calls either way.

With that in mind, I don't think we'll need a check_alias bool at all.